ambry-protocol/src/main/java/com/github/ambry/protocol/FileCopyProtocolGetMetaDataResponse.java

```diff
@@ -0,0 +1,98 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.protocol;
+
+import com.github.ambry.server.ServerErrorCode;
+import com.github.ambry.utils.Utils;
+import io.netty.buffer.PooledByteBufAllocator;
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class FileCopyProtocolGetMetaDataResponse extends Response {
+  private final int numberOfLogfiles;
+
+  private final String hostName;
+  private final List<LogInfo> logInfoList;
+  private static final short File_Copy_Protocol_Metadata_Response_Version_V1 = 1;
+  private static final int HostName_Field_Size_In_Bytes = 4;
+
+  public FileCopyProtocolGetMetaDataResponse(short versionId, int correlationId, String clientId, int numberOfLogfiles,
+      List<LogInfo> logInfoList, ServerErrorCode errorCode, String hostName) {
+    super(RequestOrResponseType.FileCopyProtocolGetMetaDataResponse, versionId, correlationId, clientId, errorCode);
+    this.numberOfLogfiles = numberOfLogfiles;
+    this.logInfoList = logInfoList;
+    this.hostName = hostName;
+  }
+
+  public static FileCopyProtocolGetMetaDataResponse readFrom(DataInputStream stream) throws IOException {
+    RequestOrResponseType type = RequestOrResponseType.values()[stream.readShort()];
+    if (type != RequestOrResponseType.FileCopyProtocolGetMetaDataResponse) {
+      throw new IllegalArgumentException("The type of request response is not compatible");
+    }
+    short versionId = stream.readShort();
+    int correlationId = stream.readInt();
+    String clientId = Utils.readIntString(stream);
+    ServerErrorCode errorCode = ServerErrorCode.values()[stream.readShort()];
+
+    if(errorCode != ServerErrorCode.No_Error) {
+      return new FileCopyProtocolGetMetaDataResponse(versionId, correlationId, clientId, -1, new ArrayList<>(), errorCode, null);
+    }
+
+    String hostName = Utils.readIntString(stream);
+    int numberOfLogfiles = stream.readInt();
+    int logInfoListSize = stream.readInt();
+    List<LogInfo> logInfoList = new ArrayList<>();
+    for (int i = 0; i < logInfoListSize; i++) {
+      logInfoList.add(LogInfo.readFrom(stream));
+    }
+    return new FileCopyProtocolGetMetaDataResponse(versionId, correlationId, clientId, numberOfLogfiles, logInfoList, errorCode, hostName);
+  }
+  protected void prepareBuffer() {
+    super.prepareBuffer();
+    Utils.serializeString(bufferToSend, hostName, Charset.defaultCharset());
+    bufferToSend.writeInt(numberOfLogfiles);
+    bufferToSend.writeInt(logInfoList.size());
```

Comment:
what is the difference between these two?

Reply:
removed redundancy.